mitosheet/mitosheet/step_performers/graph_steps/graph_duplicate.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# Copyright (c) Saga Inc.
+# Distributed under the terms of the GPL License.
+from copy import copy, deepcopy
+from typing import Any, Dict, List, Optional, Set, Tuple
+import uuid
+
+from mitosheet.state import State
+from mitosheet.step_performers.step_performer import StepPerformer
+from mitosheet.types import GraphID
+
+
+class GraphDuplicateStepPerformer(StepPerformer):
+    """
+    This steps duplicates a graph of a given graphID. 
+    """
+
+    @classmethod
+    def step_version(cls) -> int:
+        return 1
+
+    @classmethod
+    def step_type(cls) -> str:
+        return 'graph_duplicate'
+
+    @classmethod
+    def step_display_name(cls) -> str:
+        return 'Duplicated a Graph'
+    
+    @classmethod
+    def step_event_type(cls) -> str:
+        return 'graph_duplicate_edit'
+
+    @classmethod
+    def saturate(cls, prev_state: State, params: Dict[str, Any]) -> Dict[str, Any]:
+        return params
+
+    @classmethod
+    def execute( # type: ignore
+        cls,
+        prev_state: State,
+        old_graph_id: GraphID,
+        new_graph_id: GraphID,
+        **params
+    ) -> Tuple[State, Optional[Dict[str, Any]]]:
+        post_state = deepcopy(prev_state)
+
+        # Execute the step
+        graph_copy = deepcopy(post_state.graph_data[old_graph_id])
+        graph_copy["graphTabName"] = graph_copy["graphTabName"] + '_copy'
```

Comment:
Can you note here that we don't need to insist that graphTabNames are unique? Since we don't transpile them linearly, this is not an issue.